src/shape.rs

```diff
@@ -23,53 +23,37 @@ pub struct Shape {
 }
 
 impl Shape {
-    /// Common function to create a new `Shape` with calculating inner statistics.
+    /// Creates a new n-dimensional shape.
     ///
     /// # Arguments
     ///
-    /// * `num_dimensions` - Number of dimensions in the new shape.
-    /// * `dimensions` - Number of elements for each dimension.
+    /// * `dimensions` - Number of elements for each axis.
     ///
     /// # Returns
     ///
     /// A new `Shape` object.
     ///
     /// # Panics
     ///
-    /// `num_dimensions` exceeds MAX_NUM_DIMENSIONS.
-    fn new_inner(num_dimensions: usize, dimensions: [usize; MAX_NUM_DIMENSIONS]) -> Self {
+    /// The const parameter `N` is larger than `MAX_NUM_DIMENSIONS`.
+    pub fn new<const N: usize>(dimensions: [usize; N]) -> Self {
```

Comment:
I think it is better to prepare another function to create the object form a slice. Array is size-known object so that optimization can be easily applied than slices.

Reply:
Added acfd2d8